src/nprintml/label/aggregator/pcap.py

```diff
@@ -44,17 +44,28 @@ def __call__(self, compress=False, sample_size=1):
         print('Attaching labels to nPrints')
 
         labels = self.load_label(self.label_csv)
+        labels = self.rewrite_labels(labels, npts)
+
         (npt_df, missing_labels, ogns, nns) = self.attach_label(npts, labels)
 
         print('  labels attached: missing labels for:', missing_labels)
         print('    missing labels caused samples to be dropped:', (ogns - nns))
 
         return npt_df
-
-    def merge_npt(self):
-        """Merge nPrint data from multiple output files."""
-        # Load nPrint data sets and determine largest set
-        print('Loading nprints')
+    
+    def rewrite_labels(self, labels, npts):
+        new_labels = []
+        stems = {}
+        # Get stems for orignal files
+        for row in labels.itertuples():
+            stems[pathlib.PurePath(row.Index).stem] = row.Label
+
+        for row in npts.itertuples():
+            stem = pathlib.PurePath(row.Index).stem
+            if stem in stems:
+                new_labels.append((row.Index, stems[stem]))
+
+        return pd.DataFrame(new_labels, columns=['item', 'label']).set_index('item')
```

Comment:
I believe I can make out what's going on here. Regardless, some thoughts:

* this method could use a docstring and/or comment(s) – however terse
* the required labeling file format is undocumented as far as I'm aware
* the (full-path) index of `npts` established in `merge_npt` is of our own making and appears to make for unnecessary work here. this could be made the "npt" file's stem, for example.

Reply:
Docstring Done.
Labeling and documentation should be updated. I'll work on that.
nPrint file stems could be saved, though i am a bit hesitant to save only the stems in case we need them for some other reason. Ideally we would have a better matching structure so we could allow for recursive directories of PCAPs, but that may require attaching labels in a different spot (before we run nPrint)